services/src/datasets/external/nature40.rs

```diff
@@ -129,34 +131,54 @@ impl DatasetProvider for Nature40DataProvider {
         for (db, dataset) in raster_dbs.rasterdbs.iter().zip(datasets) {
             if let Ok(dataset) = dataset {
                 let (dataset, band_labels) = self.get_band_labels(dataset).await?;
+                let (dataset, time_positions) = self.get_time_positions(dataset).await?;
 
-                for band_index in 1..=dataset.raster_count() {
-                    if let Ok(result_descriptor) =
-                        raster_descriptor_from_dataset(&dataset, band_index, None)
-                    {
-                        listing.push(Ok(DatasetListing {
-                            id: DatasetId::External(ExternalDatasetId {
-                                provider_id: self.id,
-                                dataset_id: format!("{}:{}", db.name.clone(), band_index),
-                            }),
-                            name: db.title.clone(),
-                            description: format!(
-                                "Band {}: {}",
-                                band_index,
-                                band_labels
-                                    .get((band_index - 1) as usize)
-                                    .unwrap_or(&"".to_owned())
-                            ),
-                            tags: db.tags.clone(),
-                            source_operator: "GdalSource".to_owned(),
-                            result_descriptor: TypedResultDescriptor::Raster(result_descriptor),
-                            symbology: None, // TODO: build symbology
-                        }));
-                    } else {
-                        info!(
-                            "Could not create restult descriptor for band {} of {}",
-                            band_index, db.name
-                        );
+                let mut time_indexes = 1;
+                let mut time_layer = false;
+
+                if !time_positions.is_empty() {
+                    time_indexes = time_positions.len();
+                    time_layer = true;
+                }
+
+                for time_index in 0..time_indexes {
```

Comment:
- different time positions should not be separate datasets
  - we want to use one dataset which changes over time, i.e. querying the dataset with a different time should yield a different result

Reply:
DatasetProvider for Nature40 restored to original.